lib/room-app-stack.ts

```diff
@@ -24,47 +26,43 @@ export class RoomAppStack extends cdk.Stack {
   constructor(scope: cdk.Construct, id: string, props: RoomAppProps = {}) {
     super(scope, id, props);
 
-    const {
-      fromAddress,
-      domainName,
-      zoneId,
-      facebookAppId,
-      facebookAppSecret,
-      amazonClientId,
-      amazonClientSecret,
-      googleClientId,
-      googleClientSecret,
-    } = props;
+    const { domainName, zoneId } = props;
 
     const stageName = "ws";
 
+    const connectionsTable = new Connections.Table(this);
+
     const connectFn = new lambda.Function(this, "ConnectionHandler", {
       runtime: lambda.Runtime.NODEJS_14_X,
-      handler: "index.connect",
-      code: lambda.Code.fromAsset("backend/connect"),
+      handler: "connections.connectHandler",
+      code: lambda.Code.fromAsset("backend"),
       memorySize: 3000,
       environment: {
         NODE_ENV: "production",
+        CONNECTIONS_TABLE_NAME: connectionsTable.tableName,
       },
       timeout: cdk.Duration.seconds(20),
     });
     const disconnectFn = new lambda.Function(this, "DisconnectionHandler", {
       runtime: lambda.Runtime.NODEJS_14_X,
-      handler: "index.disconnect",
-      code: lambda.Code.fromAsset("backend/connect"),
+      handler: "connections.disconnectHandler",
+      code: lambda.Code.fromAsset("backend"),
       memorySize: 3000,
       environment: {
         NODE_ENV: "production",
+        CONNECTIONS_TABLE_NAME: connectionsTable.tableName,
       },
       timeout: cdk.Duration.seconds(20),
     });
     const defaultFn = new lambda.Function(this, "DefaultHandler", {
       runtime: lambda.Runtime.NODEJS_14_X,
-      handler: "index.default",
-      code: lambda.Code.fromAsset("backend/connect"),
+      handler: "connections.defaultHandler",
+      code: lambda.Code.fromAsset("backend"),
       memorySize: 3000,
       environment: {
         NODE_ENV: "production",
+        ENDPOINT: "https://d1vy5lwn12jrv5.cloudfront.net/ws/",
```

Comment:
If you move [the frontend bucket instantiation](https://github.com/douglasnaphas/madliberation/issues/343) and [distro instantiation](https://github.com/dan-muller/room-app/blob/a6ff6ed81e44550542cdb07e200fdd03d53c030b/lib/room-app-stack.ts#L112-L126) up above this `lambda.Function` instantiation, you can replace this with

```ts
        ENDPOINT: `https://${distro.distributionDomainName}/ws/`,
```

Reply:
More importantly, the `defaultHandler` will push messages to the WebSocket clients using the endpoint URL provided by API Gateway, not CloudFront.

It may seem for a moment like we have a chicken-or-the-egg problem: the Lambda function `defaultFn` has to reference the API Gateway API (to get the environment variable showing where to send messages), and the API Gateway API has to reference `defaultFn` (to specify it as the default handler).

However, you can leave out the `ENDPOINT` environment variable when you instantiate `defaultFn`, and then, after you've instantiated `webSocketApi`, you can call [addEnvironment](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.Function.html#addwbrenvironmentkey-value-options) on `defaultFn` to add the `ENDPOINT` environment variable referencing the API Gateway WebSocket API.